feat(web-analytics): Add custom bounce rate control

robbie-c opened the pull request:
## Problem

See https://posthog.slack.com/archives/C05LJK1N3CP/p1734713816519279 for context

Docs https://github.com/PostHog/posthog.com/pull/10206

## Changes

Make the bounce rate duration threshold configurable

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Added a test - see diff

robbie-c committed:
Add custom bounce rate control

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

rafaeelaudibert commented on frontend/src/scenes/settings/environment/BounceRateDuration.tsx:
```diff
@@ -0,0 +1,34 @@
+
+    return (
+        <>
+            <p>Choose how long a user can stay on a page, before the session is not a bounce.</p>
```
Let's make it clear we're talking seconds here

robbie-c replied:
Done

rafaeelaudibert commented on posthog/hogql/database/schema/sessions_v1.py:
```diff
@@ -230,7 +236,10 @@ def arg_max_merge_field(field_name: str) -> ast.Call:
                             # if session duration >= 10 seconds, not a bounce
```
Let's update the comment

Btw, we were using 10 seconds, not 30, lol, website docs are probably outdated

robbie-c replied:
https://github.com/PostHog/posthog.com/pull/10206

rafaeelaudibert commented on posthog/hogql/database/schema/sessions_v2.py:
```diff
@@ -37,6 +37,9 @@
     from posthog.models.team import Team
 
+
+DEFAULT_BOUNCE_RATE_DURATION_SECONDS = 30
```
Default was actually 10 if you look at the old line 274

robbie-c replied:
I should not try to work on a train 🙃

rafaeelaudibert commented on posthog/hogql/database/schema/sessions_v2.py:
```diff
@@ -271,7 +279,10 @@ def arg_max_merge_field(field_name: str) -> ast.Call:
                                 # if session duration >= 10 seconds, not a bounce
```
There are a bunch of comments that need to be updated

robbie-c committed:
Default was 10

robbie-c committed:
Improve copy

robbie-c committed:
Fix circular imports

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

robbie-c committed:
Fix tests

robbie-c committed:
Improve the UI to change the setting

rafaeelaudibert reviewed (approved):
LGTM!

Is this `query_modifier` a JSON object for each team? I'm asking because I don't see a DB migration

robbie-c committed:
Merge branch 'master' into add-custom-bounce-rate-control

rafaeelaudibert committed:
Merge branch 'master' into add-custom-bounce-rate-control

rafaeelaudibert committed:
refactor: Avoid changing `LemonInput`

There's a weird E2E test failing after the `LemonInput` changes, let's revert them to see if this works now
